torch/_inductor/cudagraph_utils.py

```diff
@@ -234,3 +234,20 @@ def log_data_ptr_mismatch(
                 f"input stack trace: {get_placeholder_stack_trace(placeholder)}\n"
             )
     return error_msg
+
+
+def maybe_warning_due_to_dynamic_shape(
+    fn_cache: Dict[Tuple[int, ...], Callable[..., Any]],
+    new_int_key: Any,
+):
+    num_cudagraphs = len(fn_cache.keys()) + 1
+
+    warn_msg = (
+        "CUDAGraph supports dynamic shapes by recording a new graph for each "
+        f"distinct input size. We have observed {num_cudagraphs} distinct "
+        f"sizes, including {[*fn_cache.keys(), new_int_key]}. Consider padding "
+        "inputs to a few fixed number of shapes for better performance."
+    )
```

Comment:
Could we also add something like "Set `cudagraph_dynamic_shape_warn_limit ` to None to silence this warning."